cloud/cluster/cluster.go

```diff
@@ -0,0 +1,64 @@
+package cluster
+
+import (
+	"fmt"
+	"io"
+)
+
+type NodeId string
+
+type Node interface {
+	Id() NodeId
+}
```

Comment:
nit: can Node Interface and related structs/types be in its own file.  I keep forgetting its in cluster and spend a bunch of time looking for it.